src/index.ts

```diff
@@ -0,0 +1,518 @@
+import LanguageTag from './utils/LanguageTag';
+import {
+  PluralCategory,
+  arabic,
+  eastSlavic,
+  oneOther,
+  oneTwoOther,
+  oneUpToTwoOther,
+  oneWithZeroOther,
+  other,
+  polish,
+  westSlavic,
+} from './utils/pluralization';
+
+// Define some variables for validating keys and key context.
+const CONTEXT_CHARACTERS = '[A-Za-z0-9-_]+';
+const DELIMITER = '.';
+
+const VALID_KEY_CONTEXT = new RegExp(`^${CONTEXT_CHARACTERS}$`);
+const VALID_KEY = new RegExp(`^${CONTEXT_CHARACTERS}(${DELIMITER}${CONTEXT_CHARACTERS})*$`);
+
+// Used to find variable names inside the output string.
+const INTERPOLATION_REGEX = /%\{([^}]*)\}/g;
+
+// Default locale is english.
+const DEFAULT_LOCALE = 'en';
+
+// Export PluralCategory so it is easy for others to create rules.
+export { PluralCategory };
+
+interface Aliases {
+  [propName: string]: string;
```

Comment:
According to the documentation, an indexable type can only be a string or a number.

https://www.typescriptlang.org/docs/handbook/interfaces.html#indexable-types

Also interesting:

> There are two types of supported index signatures: string and number. It is possible to support both types of indexers, but the type returned from a numeric indexer must be a subtype of the type returned from the string indexer. This is because when indexing with a number, JavaScript will actually convert that to a string before indexing into an object.

So, theoretically, I can't say a `number` type return undefined, without saying it could also be a string. It's pretty strange, perhaps I should use an actual `Map` type instead? Then I don't need these interfaces. I think I'll make that switch.

Reply:
Don't forget that using an ES6 `Map` means either including a `Map` polyfill or assuming that the code will always be run in an environment that supports `Map` natively.

Reply:
See: https://github.com/Microsoft/TypeScript/issues/3069#issuecomment-99964139

Reply:
This is a good point. I think I want to just compile to ES5 to make the payload smaller. Most browsers support that functionality and I don't think we really need to support ES3.